### PR TITLE
feature: api keys as an alternative auth mechanism

### PR DESCRIPTION
**Pull request by skeptrunedev**
image

### CODE REVIEW

**Comment by skeptrunedev** [comment]
Run the following in devtools console:

```
fetch("http://localhost:8090/api/user/set_api_key", {
    method: "GET",
    credentials: "include"
})
```

after signing in to get an API key. It will show up in the response on the devtools network tab.

File: server/src/operators/card_operator.rs [hunk 1]
```diff
@@ -545,6 +545,7 @@ pub fn get_metadata_query(
                 user_columns::username,
                 user_columns::website,
                 user_columns::visible_email,
+                user_columns::api_key_hash,
```
Comment (densumesh) [inline server/src/operators/card_operator.rs:548]: Do we want to include their api key in the return request? seems like it might be a security issue

Comment (densumesh) [inline server/src/operators/card_operator.rs:548]: Do we want their api key in the return request? seems like a security issue

Comment (skeptrunedev) [reply server/src/operators/card_operator.rs:548]: I mean the user has to be able to see their created api key some way or another.

**Review by fastfists — CHANGES_REQUESTED** [review]

File: server/src/handlers/register_handler.rs [hunk 2]
```diff
@@ -122,3 +123,14 @@ fn insert_user_from_invitation(
...
+
+pub fn generate_api_key() -> String {
+    let rng = rand::thread_rng();
+    let api_key: String = rng
+        .sample_iter(&Alphanumeric)
+        .take(32)
+        .map(char::from)
+        .collect();
+
+    api_key
```
[the remark this answers is not in the document]
Comment (skeptrunedev) [reply server/src/handlers/register_handler.rs]: Fixex

**Pull request merged**